sklearn/decomposition/tests/test_pca.py

```diff
@@ -340,13 +340,28 @@ def test_pca_inverse():
 
 
 def test_pca_validation():
-    X = [[0, 1], [1, 0]]
+    # Ensures that extreme inputs for n_components common to all solvers
+    # (less than 0 or more than the lesser dimension of the input
+    # matrix X) raise errors.
+    X = np.array([[0, 1, 0], [1, 0, 0]])
     for solver in solver_list:
         for n_components in [-1, 3]:
```

Comment:
You should use `assert_raises_regex` to check also the message return and not only `assert_raises.`